src/world_modeling/occupancy_segmentation/include/occupancy_segmentation_core.hpp

```diff
@@ -0,0 +1,377 @@
+#ifndef OCCUPANCY_SEGMENTATION_CORE_HPP_
+#define OCCUPANCY_SEGMENTATION_CORE_HPP_
+
+#include <pcl/common/centroid.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <vector>
+
+#include <tbb/blocked_range.h>
+#include <tbb/parallel_for.h>
+#include <Eigen/Dense>
+
+#define NUM_ZONES 4
+#define L_MIN 2.7
+#define L_MAX 80.0
+
+#define Z_SEED 0.5
+#define MD 0.3
+#define MH -1.1
+
+#define MIN_NUM_POINTS 10
+#define NUM_SEED_POINTS 20
+#define TH_SEEDS 0.5
+#define UPRIGHTNESS_THRESH 0.5
+
+#define NUM_RINGS_OF_INTEREST 4
+#define SENSOR_HEIGHT 1.7
+#define GLOBAL_EL_THRESH 0
```

Comment:
should be part of the Yaml config file, you can pass ROS loggers and param servers into core, that's ok

Reply:
done